File: Backend/src/main/java/com/example/bus_reservation_system/entity/Bus.java
package com.example.bus_reservation_system.entity;

import java.sql.Date;
import java.sql.Time;
import java.sql.Timestamp;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "BusTable")
public class Bus {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "bus_number", unique = true, nullable = false)
    private String busNumber;

    @Enumerated(EnumType.STRING)
    @Column(name = "bus_type", nullable = false)
    private BusType busType;

    @Column(name="arrTime",nullable=false)
    private Time arrTime;

    @Column(name="deptTime",nullable=false)
    private Time deptTime;

    @Column(name="route_id",nullable=false)
    private long route_id; 

    @Column(nullable = false)
    private String operator;

    @Column(nullable=false)
    private int price;

}

File: Frontend/src/app/Models/bus.ts
export class Bus {
    id?: number;
    busNumber: string;
    busType: string;
    arrTime: string;
    deptTime: string;
    operator: string;
    price:number;
    route_id:number;
   
    constructor(
      busNumber: string,
      busType: string,
      arrTime: string,
      deptTime: string,
      operator: string,
      price:number,
      routeId:number,
      id?: number
      
    ) {
      this.busNumber = busNumber;
      this.busType = busType;
      this.arrTime = arrTime;
      this.deptTime = deptTime;
      this.operator = operator;
      this.id = id;
      this.route_id=routeId;
      this.price=price;
    }
  }
  